Source/OxyPlot.SkiaSharp.Tests/PngAssert.cs

```diff
@@ -0,0 +1,140 @@
+        public static void AreEqual(string expected, string actual, string message, string output)
+        {
+            var expectedImage = LoadImage(expected);
+            var actualImage = LoadImage(actual);
+
+            if (expectedImage == null)
+            {
+                EnsureFolder(expected);
+                File.Copy(actual, expected);
+                Assert.Fail("File not found: {0}", expected);
+            }
+
+            if (expectedImage.GetLength(0) != actualImage.GetLength(0))
+            {
+                Assert.Fail("Expected height: {0}\nActual height:{1}\n{2}", expectedImage.GetLength(0), actualImage.GetLength(0), message);
+            }
+
+            if (expectedImage.GetLength(1) != actualImage.GetLength(1))
+            {
+                Assert.Fail("Expected width: {0}\nActual width:{1}\n{2}", expectedImage.GetLength(1), actualImage.GetLength(1), message);
+            }
+
+            var w = expectedImage.GetLength(0);
+            var h = expectedImage.GetLength(1);
+            var differences = 0;
+            var differenceImage = new OxyColor[w, h];
+            for (var i = 0; i < h; i++)
+            {
+                for (var j = 0; j < w; j++)
+                {
+                    if (!expectedImage[j, i].Equals(actualImage[j, i]))
+                    {
+                        differences++;
+                        differenceImage[j, i] = OxyColors.Red;
+                    }
+                    else
+                    {
+                        differenceImage[j, i] = actualImage[j, i].ChangeIntensity(100);
+                    }
+                }
+            }
+
+            if (differences > 0)
+            {
+                if (output != null)
+                {
+                    for (var i = 0; i < h; i++)
+                    {
+                        for (var j = 0; j < w; j++)
+                        {
+                            if (!expectedImage[j, i].Equals(actualImage[j, i]))
+                            {
+                            }
+                        }
+                    }
+
+                    EnsureFolder(output);
+                    var encoder = new PngEncoder(new PngEncoderOptions());
+                    File.WriteAllBytes(output, encoder.Encode(differenceImage));
+                }
+
+                Assert.Fail(
+                    "{0}:\nPixel differences: {1}\nExpected image: {2}\nActual image: {3}\nDiff image: {4}",
+                    message,
+                    differences,
+                    Path.GetFullPath(expected),
+                    Path.GetFullPath(actual),
+                    Path.GetFullPath(output));
+            }
+        }
+
+        /// <summary>
+        /// Ensures that the folder for the specified path exists.
+        /// </summary>
+        /// <param name="path">The path to a file.</param>
+        private static void EnsureFolder(string path)
```

Comment:
I'd prefer a name which indicates we are creating the directory for the given file, rather than ensuring the given directory exists. Something like `EnsureDirectoryForFile` or something, I don't have any especially good suggestions.

Err... just realised that boat sailed in 2014. Nevermind ;)

Reply:
Yeah... I pretty much just copy&pasted this file, so I'd suggest we leave this as is for now. Maybe we can find a way in the future to share this with `OxyPlot.Wpf.Tests`.